client/knowledge_base_search/src/App.js

```diff
@@ -112,6 +132,30 @@ class App extends Component {
       : Promise.resolve(enriched_results);
   }
 
+  shuffleQuestions(questions) {
+    const allQuestions = questions.slice(0);
+    let shuffledQueries = [];
+
+    for (let i = 0; i < questions.length; i++) {
+      let questionIndex = Math.floor(Math.random() * allQuestions.length);
+      shuffledQueries.push(allQuestions.splice(questionIndex, 1)[0]);
+    }
+
+    return shuffledQueries;
+  }
+
+  onQuestionClick = (query) => {
```

Comment:
Might be helpful to establish a pattern with props that are functions and their named handlers.

Example:
Prop name: `onQuestionClick`
Prop value: `this.handleQuestionClick`